Update docs with additional information on CCF Raft (#3778)
microsoft/CCF: doc/architecture/consensus/index.rst

CFT Consensus Protocol
----------------------

The crash fault tolerant implementation in CCF is based on `Raft <https://raft.github.io/>`_. The key differences between the original Raft protocol (as described in the `Raft paper <https://raft.github.io/raft.pdf>`_), and CCF Raft are as follows:

> * Transactions in CFF Raft are not considered to be committed until a subsequent signed transaction has been committed. More information can be found :doc:`here </consensus/merkle_tree>`. Transactions in the ledger before the last signed transactions are discarded during leader election.

Reviewer: ```suggestion
* Transactions in CCF Raft are not considered to be committed until a subsequent signed transaction has been committed. More information can be found :doc:`here </consensus/merkle_tree>`. Transactions in the ledger before the last signed transactions are discarded during leader election.
```

> * As well as Raft's two-phase reconfiguration protocol, CCF also supports one-phase reconfiguration. You can find more information on the Raft implementation of reconfiguration in CCF :doc:`here <1tx-reconfig>`. Note that CCF Raft does not support node restart. If a node fails and starts again, it must rejoin Raft via reconfiguration.

Reviewer 2: ```suggestion
* As well as Raft's two-phase reconfiguration protocol, CCF also supports one-phase reconfiguration. You can find more information on the Raft implementation of reconfiguration in CCF :doc:`here <1tx-reconfig>`. Note that CCF Raft does not support node restart as the unique identity of each node is tied to the node process launch. If a node fails and is replaced, it must rejoin Raft via reconfiguration.
```

I think would be slightly clearer. 

Also, the wording of this sentence suggests that the two-phase reconfiguration protocol is the default reconfiguration protocol, but it's actually the opposite (1tx is default and currently the one we recommend for production deployments) so it may be worth switching the order around.

* In CCF Raft, clients receive an acknowledgement with a :term:`Transaction ID` (view and sequence number) before the transaction has been replicated to Raft's ledger. The client can later use this transaction ID to verify that the transaction has been committed by Raft.

> * CCF Raft uses an additional mechanism so a newly elected leader can more efficiently determine the current state of a follower's ledger when the two ledgers have diverged. This enables the leader to bring the follower up to date more quickly. CFF Raft also batches appendEntries messages.

Reviewer: ```suggestion
* CCF Raft uses an additional mechanism so a newly elected leader can more efficiently determine the current state of a follower's ledger when the two ledgers have diverged. This enables the leader to bring the follower up to date more quickly. CCF Raft also batches appendEntries messages.
```
